### PR TITLE
Updating OAuth 2.0 documentation

### PR DESCRIPTION
**Pull request by jsanchez-slingr**
## Summary

This pull request updates the documentation for the Single Sign-On (SSO) feature for OAuth. Key changes include:

- Replacing instances of "URI" with "URL" throughout the documentation.
- Splitting the user processing script for better clarity and standardization.
- Introducing two methods for access token retrieval:
    Access Token Retrieval Method: Choose between "URL" or "Script".
    Access Token URL: A static endpoint from which the access token is retrieved.
    Access Token Script: Custom code for access token retrieval.

- Updating the "User Mapping Script" section to provide a more standard example.
- The "Access Token Script" documentation now references the HTTP package or service.
- The documentation will refer to the process as "OAuth 2.0"

## Motivation

These updates aim to clarify the access token retrieval process and improve the overall usability of the documentation. By standardizing examples and methods, we support a wider range of use cases and enhance user understanding, leading to more effective implementation of the SSO feature.

## Checks

- [ ] Read [Create a Pull Request](https://getdoks.org/docs/contributing/how-to-contribute/#create-a-pull-request)
- [ ] Supports all screen sizes (if relevant)
- [ ] Supports both light and dark mode (if relevant)
- [ ] Passes `npm run test`

___

### **PR Type**
Documentation

___

### **Description**
- Updated the OAuth 2.0 documentation to replace "URI" with "URL" for consistency.
- Enhanced the explanation of the access token retrieval process, including the introduction of two methods: URL and Script.
- Provided detailed examples for implementing access token scripts and user mapping scripts, improving clarity and usability.
- Improved the overall structure and flow of the documentation to better support OAuth 2.0 implementation.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Documentation</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>single-sign-on.md</strong><dd><code>Update OAuth 2.0 documentation with improved examples and clarity</code></dd></summary>
<hr>

content/en/dev-reference/security/single-sign-on.md

<li>Updated terminology from "URI" to "URL".<br> <li> Enhanced documentation for OAuth 2.0 access token retrieval.<br> <li> Added examples for access token script and user mapping script.<br> <li> Improved clarity on OAuth 2.0 processes and parameters.<br>

</details>

  </td>
  <td><a href="https://github.com/slingr-stack/slingr-docs/pull/143/files#diff-1929ad6e51628c70a31d70214313860038803fd9cf04c2ed83af2c72944f0738">+68/-60</a>&nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**: Comment `/help "your question"` on any pull request to receive relevant information

### CODE REVIEW

File: content/en/dev-reference/security/single-sign-on.md [hunk 1]
```diff
@@ -529,78 +511,104 @@ When you register an application, a unique and permanent secret is automatically
...
+### Access token script
 
-##### Samples
+This section will be available when the `Script` option has been chosen under the `Access Token Retrieval Method.` You will need to obtain the access token manually and return it under the key name `accessToken` (see [HTTP Service](https://github.com/slingr-stack/http-service) and [HTTP Package](https://github.com/slingr-stack/http-package)). The `code` obtained as part of the authorization flow will be provided as a parameter, also you will be able to access the configuration defined for the provided under the `config` parameter. For example:
 
   ```js
-    // get access token
-  const accessTokenResponse = svc.http.post({
+    const accessTokenResponse = svc.http.post({
```
Comment (pasaperez-slingr) [inline content/en/dev-reference/security/single-sign-on.md:540]: Here let's use the package
```suggestion
    const accessTokenResponse = pkg.http.post({
```

File: content/en/dev-reference/security/single-sign-on.md [hunk 2]
```diff
@@ -529,78 +511,104 @@ When you register an application, a unique and permanent secret is automatically
...
+
+- **`accessToken`**: This parameter is the access token received from the user's authentication process.
+- **`config`**: This is a JSON object that contains all the configuration settings defined for the identity provider, including `client_id`, `client_secret`, `scope`, `oauthRedirectUri` (which corresponds to the OAuth redirect URI), and `token_url` (which corresponds to the Access Token URL).
+- **`userRecord`**: This parameter represents the user being authenticated.
+
+A call should be made to the provider's API to retrieve the user's data (see [HTTP Service](https://github.com/slingr-stack/http-service) and [HTTP Package](https://github.com/slingr-stack/http-package)). The script should first check if the user already exists by searching for their email. If the user exists, return their record along with the associated firstName and lastName. If the user does not exist, the developer must ensure that the fields `firstName`, `lastName`, and `email` are filled out; otherwise, the user record cannot be saved, as these fields are required. Finally, the script must return the userRecord.
+
+  ```js
+  const userDataResponse = svc.http.get({
```
Comment (pasaperez-slingr) [inline content/en/dev-reference/security/single-sign-on.md:569]:
```suggestion
  const userDataResponse = pkg.http.get({
```

**Review by pasaperez-slingr — CHANGES_REQUESTED** [review]

**Comment by sebasmoyano** [comment]
#approve

**Comment by pasaperez-slingr** [comment]
Looks good, thanks Joaquín!

**Pull request merged**